File: selenium-e2e/src/test/java/dev/dsbon/realworld/pages/OnboardingDialog.java
package dev.dsbon.realworld.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

/**
 * The modal first-run wizard shown to a user with no bank account. It blocks
 * every other interaction until completed, so any test that registers a fresh
 * account has to expect it.
 */
public class OnboardingDialog extends BasePage {

  private static final By DIALOG = testId("user-onboarding-dialog");
  private static final By TITLE = testId("user-onboarding-dialog-title");
  private static final By NEXT = testId("user-onboarding-next");

  public OnboardingDialog(WebDriver driver) {
    super(driver);
  }

  public String awaitTitle() {
    visible(DIALOG);
    return textOf(TITLE);
  }

  public boolean isPresent() {
    return isPresent(DIALOG);
  }

  public void next() {
    click(NEXT);
  }
}

File: cypress-e2e/cypress/pages/OnboardingDialog.ts
import { SEL } from "../support/selectors";

/**
 * The first-run wizard shown to a user who has no bank account yet.
 *
 * It is modal, so it blocks every other interaction until it is completed or the
 * user signs out. Any test that registers a fresh account has to account for it.
 */
export class OnboardingDialog {
  shouldBeVisible(): this {
    cy.get(SEL.onboarding.dialog).should("be.visible");
    cy.get(SEL.onboarding.title).should("contain.text", "Get Started");
    return this;
  }

  next(): this {
    cy.get(SEL.onboarding.next).click();
    return this;
  }

  shouldNotExist(): this {
    cy.get(SEL.onboarding.dialog).should("not.exist");
    return this;
  }
}

export const onboarding = new OnboardingDialog();
